### PR TITLE
[EventGrid] Adjust README verbiage in preparation for UX pilot study with feedback

### PR DESCRIPTION
**Pull request by KieranBrantnerMagee**
- improve clarity of "host" vs topic endpoint (which is the verbiage actually seen in the portal)
- Add breadcrumbs to generate_sas and sas based auth
- change api key verbiage to access key

### CODE REVIEW

**Review by elraikhm — APPROVED** [review]

**Review by rakshith91 — APPROVED** [review]

File: sdk/eventgrid/azure-eventgrid/dev_requirements.txt [hunk 1]
```diff
@@ -2,4 +2,4 @@
 -e ../../../tools/azure-sdk-tools
 -e ../../core/azure-core
 -e ../../identity/azure-identity
--e ../azure-mgmt-eventgrid
+azure-mgmt-eventgrid==3.0.0rc8
```
Comment (lmazuel) [inline sdk/eventgrid/azure-eventgrid/dev_requirements.txt:5]: Why?

**Comment by rakshith91** [comment]
Closing this in favor of https://github.com/Azure/azure-sdk-for-python/pull/16124

File: sdk/eventgrid/azure-eventgrid/azure/eventgrid/_models.py [hunk 2]
```diff
@@ -22,15 +22,15 @@ class EventMixin(object):
...
+        Otherwise, leaves it as-is.
 
         :param str event_type: The event_type of the EventGridEvent object or the type of the CloudEvent object.
         """
         # if system event type defined, set event.data to system event object
         try:
             event.data = (_event_mappings[event_type]).deserialize(event.data)
-        except KeyError: # else, if custom event, then event.data is dict and should be set to None
-            event.data = None
+        except KeyError: # else, if custom event, then event.data is dict and is left as-is.
```
Comment (rakshith91) [inline sdk/eventgrid/azure-eventgrid/azure/eventgrid/_models.py:32]: will tackle this in https://github.com/Azure/azure-sdk-for-python/issues/16125

**Pull request closed without merging**